AK/Types.h

```diff
@@ -67,23 +67,20 @@ using nullptr_t = decltype(nullptr);
 
 static constexpr FlatPtr explode_byte(u8 b)
 {
-#if ARCH(I386)
-    return (u32)b << 24 | (u32)b << 16 | (u32)b << 8 | (u32)b;
-#else
-    return (u64)b << 56 | (u64)b << 48 | (u64)b << 40 | (u64)b << 32 | (u64)b << 24 | (u64)b << 16 | (u64)b << 8 | (u64)b;
-#endif
+    FlatPtr value = b;
+    if constexpr (sizeof(FlatPtr) == 4)
+        return value << 24 | value << 16 | value << 8 | value;
+    else
+        return value << 56 | value << 48 | value << 40 | value << 32 | value << 24 | value << 16 | value << 8 | value;
```

Comment:
I know this is a slight change in behavior, but I don't like how it assumes that FlatPtr is 8 bytes if it isn't 4 bytes. How about doing something like:
```suggestion
    FlatPtr value = b;
    if constexpr (sizeof(FlatPtr) == 4)
        return value << 24 | value << 16 | value << 8 | value;
    else if constexpr (sizeof(FlatPtr) == 8)
        return value << 56 | value << 48 | value << 40 | value << 32 | value << 24 | value << 16 | value << 8 | value;
```
Where leaving the last "else" branch empty will cause a compilation failure if FlatPtr isn't the expected size?

Reply:
My thought was that FlatPtr is either 4 or 8 bytes, since it's defined as "sizeof(void*) == 8 ? u64 : u32", if someone changes that to a smaller size, the branch will hit the 8 byte one, which will produce an error for shifting more bits than exists in the type.
Making it bigger than u64...I don't think so lol